### DISCUSSION TITLE
A very interactive UI

### DISCUSSION THREAD

**Discussion by snipeart007** [pystardust/ani-cli#963] [Ideas] [answered]
Lead: @port19x
Maintainers: @CoolnsX, @justchokingaround, @Derisis13

Guys, I know you're already doing a lot. But, I have a suggestion. Why don't we try and use a TUI like [Mangal](https://github.com/metafates/mangal)

It's possible to just fork it and change it into a anime scraper.

Even though I'm not good at programming, I think it's possible for you guys to do it.

Mangal is already quite extensible and this quality is very useful for anime lovers as anime websites don't seem to hold on for long. They either shut down or stop working due to increased traffic.

**Comment by port19x** [comment, marked as answer]
This would indeed be possible.

Mangal uses the charmbracelet stack for his TUI [(source)](https://github.com/metafates/mangal/blob/main/go.mod)
Charmbracelet exposes the relevant and necessary subset of their tooling to shell via [gum](https://github.com/charmbracelet/gum)
Chokerman and I have written small scripts with gum before.
I'll raise the concern with the team.

Some considerations:
- gum is a lot more niche than fzf, possibly hurting portability due to it being unpackad in many places. [(source)](https://repology.org/project/gum-shell-tool/versions) [scoop version](https://repology.org/project/gum-shell-tool/versions)
- gum is packaged in arch linux, termux and scoop. => Natively packaged for 3/4 of our target plattforms
- gum is prettier, no question
- Both fzf and gum are a firm departure from the previous native UI

Video hosting is (very) expensive. Primarily in terms of storage and bandwidth.
Most if not all the piracy anime websites are a loosing buisness or at least barely profitable.
This makes that area of piracy especially volatile, or at least a lot more than manga.

**Reply by snipeart007** [reply]
I expect you to convey what the team thinks about this.
Use this thread to do so, please.

**Reply by port19x** [reply]
We're considering it as a 4.1 release

**Reply by snipeart007** [reply]
That's good to hear.

**Reply by snipeart007** [reply]
Has there been any progress. Atleast from the planning perspective. Would be pleased to know more.
